docs/tools/testing/icetea.md

```diff
@@ -0,0 +1,188 @@
+        `STATIC_DEPS=true sudo pip install lxml`
+
+* Windows
+    * python-lxml installation is problematic on Windows, since
+    it usually requires build tools. It can, however, be installed
+    from pre-built binaries.
+        * Search the internet for a binary for you system.
+        * Navigate to the directory where you downloaded the
+        binary and install it with `pip install <insert_file_name>`
+    * You can also follow [these instructions](http://lxml.de/installation.html#installation)
+    instead.
+
+#### Optional
+
+* If you wish to decorate your console log with all kinds of colors,
+install the coloredlogs module by using pip: `pip install coloredlogs`
+    * There have been issues with coloredlogs installation on Windows.
+     We might switch to a different module at some point, to enable
+     colored logging on Windows as well.
+
+### Installation
+
+`> pip install icetea`
+
+### Usage
+
+To print the help page:
+
+`icetea --help`
+
+To list all local test cases from the examples subfolder:
+
+`icetea --list --tcdir examples`
+
+To print the Icetea version:
+
+`icetea --version`
+
+#### Typical use
+
+All of the commands described below might also require other options,
+depending on the test case.
+
+**Running test cases by using the tc argument**
+
+`> icetea --tc <test case name> --tcdir <test case search path>`
+
+To run all existing test cases from the `examples` folder:
+
+`> icetea --tc all --tcdir examples`
+
+**Running an example test case with hardware**
+
+In this example, we assume that a compatible board has been connected
+to the computer and that an application binary for the board is available.
+The referred test case is available in [the Icetea GitHub repository](https://github.com/ARMmbed/icetea/blob/master/examples/test_cmdline.py).
+
+`> icetea --tc test_cmdline --tcdir examples --type hardware --bin <path to a binary>`
+
+**Using metadata filters**
+
+To run all test cases with test-type regression in the metadata:
+
+`> icetea --testtype regression --tcdir <test case search path>`
+
+The following metadata filters are available:
+* test type (`--testtype`)
+* test subtype (`--subtype`)
+* feature (`--feature`)
+* test case name (`--tc`)
+* tested component (`--component`)
+* test case folder (`--group`)
+
+**Running a pre-made suite**
+
+Icetea supports a suite-file that describes a number of test cases
+in `json` format:
+
+`> icetea --suite <suite file name> --tcdir <test case search path> --suitedir <path to suite directory>`
+
+**Enabling debug level logging**
+
+Use `-v` or `-vv` arguments to control logging levels. `-v` increases the framework's logging level
+to debug (default is info), the level of logging in
+certain plugins, and external components to info (default is warning).
+--vv increases the level of logging on all Icetea loggers to debug.
+
+**Further details**
+
+For further details on any of these features, see our documentation.
```

Comment:
This should include a link to our documentation, so: "...see our [documentation](link to our documentation).

Reply:
There used to be a link here. I was told to remove it since the link is already at the top of the file.

Reply:
Who said to remove it? Whenever we say "see our documentation," we always link. Unless...is this the documentation (this page)? Or was it linking to another page?

Reply:
@peknis01 . I can add the links if needed.